nixos/modules/services/networking/wstunnel.nix

```diff
@@ -0,0 +1,369 @@
+{ config, lib, options, pkgs, utils, ... }:
+with lib;
+let
+  cfg = config.services.wstunnel;
+  attrsToArgs = attrs: utils.escapeSystemdExecArgs (
+    mapAttrsToList
+    (name: value: if value == true then "--${name}" else "--${name}=\"${value}\"")
+    attrs
+  );
+  commonOptions = {
+    package = mkPackageOptionMD pkgs "wstunnel" {};
+
+    autoStart = mkOption {
+      description = mdDoc "Whether this tunnel server should be started automatically.";
+      type = types.bool;
+      default = true;
+    };
+
+    extraArgs = mkOption {
+      description = mdDoc "Extra command line arguments to pass to `wstunnel`. Attributes of the form `argName = true;` will be translated to `--argName`, and `argName = \"value\"` to `--argName=value`.";
+      type = with types; attrsOf (either str bool);
+      default = {};
+      example = {
+        "someNewOption" = true;
+        "someNewOptionWithValue" = "someValue";
+      };
+    };
+
+    verboseLogging = mkOption {
+      description = mdDoc "Enable verbose logging.";
+      type = types.bool;
+      default = false;
+    };
+
+    environmentFile = mkOption {
+      description = mdDoc "Environment file to be passed to the systemd service. Useful for passing secrets to the service.";
+      type = types.nullOr types.path;
+      default = null;
+      example = "/var/lib/secrets/wstunnelSecrets";
+    };
+  };
+
+  serverSubmodule = { config, ...}: {
+    options = commonOptions // {
+      listenAddress = mkOption {
+        description = mdDoc "Address to listen on.";
+        type = types.str;
+        default = "0.0.0.0";
+      };
+
+      port = mkOption {
+        description = mdDoc "Port to listen on. Setting this to a value below 1024 will also give the process the required `CAP_NET_BIND_SERVICE` capability.";
+        type = types.port;
+        default = if config.enableHTTPS then 443 else 80;
+        defaultText = literalExpression "if enableHTTPS then 443 else 80";
+      };
+
+      restrictTo = mkOption {
+        description = mdDoc "Accepted traffic will be forwarded only to this service. Set to `null` to disable this option.";
```

Comment:
My reasoning here was to force the user to think about whether they *really* want to let the wstunnel instance forward connections to arbitrary servers, because they might be implicitly relying on the protocol layer above wstunnel (for instance a wireguard vpn being tunneled through wstunnel) to take care of the authentication.

Reply:
sounds good in principle, but I think it's important to preserve a close resemblance between the NixOS module options and the original command line args to prevent confusion